src/transformers/training_args.py

```diff
@@ -1108,7 +1108,7 @@ def __post_init__(self):
             self.framework == "pt"
             and is_torch_available()
             and (self.device.type != "cuda")
-            and not (self.device.type == "xla" and "GPU_NUM_DEVICES" in os.environ)
+            and not (self.device.type == "xla" and xm.xla_real_devices([self.device])[0].split(":")[0] == "GPU")
```

Comment:
`xm` would be undefined if `is_torch_tpu_available` is False, but it's fine if device type won't be `xla` in this case. I'm not sure if it's guaranteed tho.

Reply:
Good point. I think "xm" is guaranteed to be imported if the device type is "xla". I just removed torch_xla and tested with native pytorch DDP and it works.

Reply:
Still, let's wrap this in a function `get_xla_device` which will return None, GPU or TPU to be sure?